### ISSUE TITLE
stopping containers with systemd: netplugin failed with no error message

### ISSUE THREAD

**Issue by timonegk** [containers/libpod#4759]
**Is this a BUG REPORT or FEATURE REQUEST? (leave only one on its own line)**

/kind bug

**Description**
Sometimes the container teardown fails with the error message `netplugin failed with no error message`. When this happens, the iptables rules for the container are not removed and the used ports can therefore not be reused.

**Steps to reproduce the issue:**

1. Use a systemd service to control a podman container

2. Set ExecStart to run a container that exposes a port. In my case it is started with `-a stdout -a stderr --rm --name containername -p 3015:3005 --volume ... image`.

3. Set ExecStop to stop the container (in my case `podman stop containername`)

4. Start and stop the service.

**Describe the results you received:**
```
Dez 29 17:59:24 host podman[13992]: time="2019-12-29T17:59:24+01:00" level=debug msg="Cleaning up container 872bd4c7f2f02e83b79d21792a31648c262502965c830df5d8a243d89cb52b0d"
Dez 29 17:59:24 host podman[13992]: time="2019-12-29T17:59:24+01:00" level=debug msg="Tearing down network namespace at /var/run/netns/cni-6a9ecf3d-ea78-4c43-2548-7eadb2eb34e8 for container 872bd4c7f2f02e83b79d21792a31648c262502965c830df5d8a243d89cb52b0d"
Dez 29 17:59:24 host podman[13992]: time="2019-12-29T17:59:24+01:00" level=info msg="Got pod network &{Name:containername Namespace:containername ID:872bd4c7f2f02e83b79d21792a31648c262502965c830df5d8a243d89cb52b0d NetNS:/var/run/netns/cni-6a9ecf3d-ea78-4c43-2548-7eadb2eb34e8 Networks:[] RuntimeConfig:map[podman:{IP: PortMappings:[{HostPort:3015 ContainerPort:3005 Protocol:tcp HostIP:}] Bandwidth:<nil> IpRanges:[]}]}"
Dez 29 17:59:24 host podman[13992]: time="2019-12-29T17:59:24+01:00" level=info msg="About to del CNI network podman (type=bridge)"
Dez 29 17:59:24 host podman[13992]: time="2019-12-29T17:59:24+01:00" level=error msg="Error deleting network: netplugin failed with no error message"
Dez 29 17:59:24 host podman[13992]: time="2019-12-29T17:59:24+01:00" level=error msg="Error while removing pod from CNI network \"podman\": netplugin failed with no error message"
Dez 29 17:59:24 host podman[13992]: time="2019-12-29T17:59:24+01:00" level=error msg="unable to cleanup network for container 872bd4c7f2f02e83b79d21792a31648c262502965c830df5d8a243d89cb52b0d: \"error tearing down CNI namespace configuration for container 872bd4c7f2f02e83b79d21792a31648c262502965c830df5d8a243d89cb52b0d: netplugin failed with no error message\""
Dez 29 17:59:24 host podman[13992]: time="2019-12-29T17:59:24+01:00" level=debug msg="unmounted container \"872bd4c7f2f02e83b79d21792a31648c262502965c830df5d8a243d89cb52b0d\""
Dez 29 17:59:24 host podman[13992]: time="2019-12-29T17:59:24+01:00" level=debug msg="Successfully cleaned up container 872bd4c7f2f02e83b79d21792a31648c262502965c830df5d8a243d89cb52b0d"
Dez 29 17:59:24 host podman[13992]: time="2019-12-29T17:59:24+01:00" level=debug msg="Container 872bd4c7f2f02e83b79d21792a31648c262502965c830df5d8a243d89cb52b0d storage is already unmounted, skipping..."
Dez 29 17:59:25 host podman[13992]: 2019-12-29 17:59:25.191941399 +0100 CET m=+24.655802916 container remove 872bd4c7f2f02e83b79d21792a31648c262502965c830df5d8a243d89cb52b0d (image=registry.mydomain.com/myimage:latest, name=containername)
Dez 29 17:59:25 host podman[13992]: time="2019-12-29T17:59:25+01:00" level=error msg="Error forwarding signal 15 to container 872bd4c7f2f02e83b79d21792a31648c262502965c830df5d8a243d89cb52b0d: container has already been removed"
Dez 29 17:59:25 host podman[13992]: time="2019-12-29T17:59:25+01:00" level=error msg="Error forwarding signal 18 to container 872bd4c7f2f02e83b79d21792a31648c262502965c830df5d8a243d89cb52b0d: container has already been removed"
```

**Describe the results you expected:**
The container should be stopped successfully and the iptable rules should be removed.

**Additional information you deem important (e.g. issue happens only occasionally):**
The issue seems to occur only when using a systemd service, executing the commands on the command line works.

**Output of `podman version`:**
```
Version:            1.6.2
RemoteAPI Version:  1
Go Version:         go1.10.4
OS/Arch:            linux/amd64
```

**Output of `podman info --debug`:**
```
debug:
  compiler: gc
  git commit: ""
  go version: go1.10.4
  podman version: 1.6.2
host:
  BuildahVersion: 1.11.3
  CgroupVersion: v1
  Conmon:
    package: 'conmon: /usr/bin/conmon'
    path: /usr/bin/conmon
    version: 'conmon version 2.0.3, commit: unknown'
  Distribution:
    distribution: debian
    version: "10"
  MemFree: 6982946816
  MemTotal: 50631647232
  OCIRuntime:
    name: runc
    package: 'runc: /usr/sbin/runc'
    path: /usr/sbin/runc
    version: |-
      runc version 1.0.0~rc6+dfsg1
      commit: 1.0.0~rc6+dfsg1-3
      spec: 1.0.1
  SwapFree: 0
  SwapTotal: 0
  arch: amd64
  cpus: 24
  eventlogger: journald
  hostname: fs4
  kernel: 4.19.0-6-amd64
  os: linux
  rootless: false
  uptime: 1083h 31m 5.62s (Approximately 45.12 days)
registries:
  blocked: null
  insecure: null
  search: null
store:
  ConfigFile: /etc/containers/storage.conf
  ContainerStore:
    number: 7
  GraphDriverName: vfs
  GraphOptions: {}
  GraphRoot: /var/lib/containers/storage
  GraphStatus: {}
  ImageStore:
    number: 10
  RunRoot: /var/run/containers/storage
  VolumePath: /var/lib/containers/storage/volumes
```

**Package info (e.g. output of `rpm -q podman` or `apt list podman`):**

```
podman/bionic,now 1.6.2-1~ubuntu18.04~ppa1 amd64 [installed]
```

**Comment by vrothberg** [comment]
Hi @timonegk, thanks for opening the issue and apologies for not seeing it earlier; it must have slipped through the Christmas break.

Can you update to the latest Podman and see if the error is still happening? For running Podman in systemd services I highly recommend following what https://www.redhat.com/sysadmin/podman-shareable-systemd-services suggests which is what `podman generate systemd --new $CTR` is doing now as well.

**Comment by sfhardman** [comment]
For me changing the `ExecStop` command from `podman stop` to `podman rm -f` resolved the issue as a workaround (Podman 1.6.4)
